bin/ansible-connection

```diff
@@ -295,7 +295,7 @@ def main():
         sys.exit("FAIL: %s" % e)
 
     ssh = connection_loader.get('ssh', class_only=True)
-    cp = ssh._create_control_path(pc.remote_addr, pc.port, pc.remote_user)
+    cp = ssh._create_control_path(pc.remote_addr, str(pc.port) + pc.connection, pc.remote_user)
```

Comment:
First avoid `str`, use `to_native` as it is py2/py3 compatible.
Second .. .i would extend the function to take a 4th 'optional' parameter to the hash instead of trying to fake it here

In the end we want to move this function from the ssh plugin to a more common spot, but that can wait